python/xir/program.py

```diff
@@ -50,37 +50,63 @@ def __init__(self, name: str, params: Union[List, Dict], wires: Tuple, use_float
         self._params = params
         self._wires = wires
 
+        self._is_adjoint = False
+        self._ctrl_wires = None
+
         self._use_floats = use_floats
 
     def __str__(self):
+        """Serialized string representation of a Statement"""
         if isinstance(self.params, dict):
             params = [f"{k}: {v}" for k, v in self.params.items()]
         else:
             params = [str(p) for p in self.params]
         params_str = ", ".join(params)
+        if params_str != "":
+            params_str = "(" + params_str + ")"
 
         wires = ", ".join([str(w) for w in self.wires])
 
-        if params_str == "":
-            return f"{self.name} | [{wires}]"
-        return f"{self.name}({params_str}) | [{wires}]"
+        modifier_str = ""
+        if self.ctrl_wires is not None:
```

Comment:
This also works:
```suggestion
        if self.ctrl_wires:
```
That said, I'm perfectly happy with the current implementation too.

Reply:
I think it's generally better to be specific here, since the implicit casting to a boolean isn't always reliable (it should be for tuples though). It's actually recommended in [PEP 8](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) as well, although in this case both would work.

Reply:
Sure, although I think this comes down to the expected behaviour in the case where
```python
self.ctrl_wires is ()
```
Specifically, the result of the current implementation is
```python
modifier_str = "ctrl[] "
```
while the outcome of the suggested implementation is
```python
modifier_str = ""
```

Reply:
That's a good point! 👍